src/System.Management.Automation/engine/CommandDiscovery.cs

```
@@ -340,12 +341,18 @@ private CommandProcessorBase CreateScriptProcessorForSingleShell(ExternalScriptI
         // #Requires -PSVersion
         // #Requires -PSEdition
         // #Requires -Module
-        internal static void VerifyScriptRequirements(ExternalScriptInfo scriptInfo, ExecutionContext context)
+        internal static void VerifyScriptRequirements(ExternalScriptInfo scriptInfo, ExecutionContext context, bool forCompletion = false)
         {
-            VerifyElevatedPrivileges(scriptInfo);
-            VerifyPSVersion(scriptInfo);
-            VerifyPSEdition(scriptInfo);
-            VerifyRequiredModules(scriptInfo, context);
+            // No point in checking these requirements if failures will be ignored
+            // VerifyRequiredModules will attempt to load the required modules which is useful.
+            if (!forCompletion)
+            {
+                VerifyElevatedPrivileges(scriptInfo);
+                VerifyPSVersion(scriptInfo);
+                VerifyPSEdition(scriptInfo);
+            }
+
+            VerifyRequiredModules(scriptInfo, context, forCompletion);
```

Maybe try-catch and ignore exceptions for completion scenario?

---

There is already a try catch in the pseudo binder that starts this whole chain of events.

---

VerifyRequiredModules already have:
```
if (!forCompletion && error is not null)
                    {
                    {
                        ScriptRequiresException scriptRequiresException =
                        ScriptRequiresException scriptRequiresException =
                            new ScriptRequiresException(
                            new ScriptRequiresException(
                                scriptInfo.Name,
                                scriptInfo.Name,
                                new Collection<string> { requiredModule.GetRequiredModuleNotFoundVersionMessage() },
                                new Collection<string> { requiredModule.GetRequiredModuleNotFoundVersionMessage() },
                                "ScriptRequiresMissingModules",
                                "ScriptRequiresMissingModules",
                                false,
                                false,
                                error);
                                error);
                        throw scriptRequiresException;
                        throw scriptRequiresException;
                    }
                    }
```

Which prevents throwing if it's for completion.